src/kudu/cfile/cfile_reader.cc

```diff
@@ -546,7 +546,7 @@ Status CFileReader::ReadBlock(const IOContext* io_context,
     scratch.Swap(&decompressed_scratch);
 
     // Set the result block to our decompressed data.
-    block = Slice(buf, uncompressed_size);
+    block = scratch.as_slice();
```

Comment:
```
    buf = scratch.get();
    block = scratch.as_slice();
```

Reply:
Didn't you miss this:

  Slice block(buf, data_size);
  ...
  CompressedBlockDecoder uncompressor(codec_, cfile_version_, block);
  ....

?